### PR TITLE
Get free book from newsletter

### PR DESCRIPTION
**Pull request by juzim**
Downloads the free ebook from the link contained in the newsletter

### CODE REVIEW

**Comment by juzim** [comment]
This branch also contains the other recent changes to avoid conflicts.

File: README.md [hunk 1]
```diff
@@ -272,6 +273,17 @@ Alternatively you can pull from [Docker Hub](https://hub.docker.com/r/kuchy/pack
...
+### cronjob
+Add this to your crontab to run the job daily at 9:
+```
+00 09 * * * /usr/bin/python PATH_TO_CRAWLER/script/spider.py --config PATH_TO_CRAWLER/config/prod.cfg --all --extras --notify join >> /tmp/packtpub.log 2>&1
+```
+
+### free ebook from the newsletter
+The script also downloads the free ebook from the weekly newsletter. The url (https://goo.gl/kUciut) is generated by an Google apps script which parses the mail. If you want to use your own source, just adapt url.bookFromNewsletter in the config. The url should point to an csv containing a single line with the url (no semicolon).
+You can also [clone the script](https://docs.google.com/spreadsheets/d/1jN5gV45uVkE0EEF4Nb-yVNfIr3o8OoiVveUZJRMiLFw) to use your own Gmail account. Just subscribe to the newsletter (found on the bottom of https://www.packtpub.com) and create a filter to tag your mails accordingly.
+
```
Comment (niqdev) [inline README.md]: That's ok, as I asked yesterday, is the google app something that you have done or where it comes from? Do you have the source code? I'd really like to see it. Is something that can be added to this project?

Comment (juzim) [inline README.md]: You can see the full code when you clone the spreadsheet, I will make this more clear in the readme.

**Comment by niqdev** [comment]
@juzim Do you have the possibility to verify the conflicts?

**Comment by juzim** [comment]
Not sure what's your position on rebase vs merge, I hope it's ok this way.

**Comment by niqdev** [comment]
I checked the 3 pull requests, I saw that you did it from different branches, should I merge just the last one i.e. #36 or do you suggest to merge all in order? The problem is that now there is no free ebook and I can't run the script!

**Comment by niqdev** [comment]
By the way..thanks, you have done a great job!

**Comment by juzim** [comment]
Yeah, I messed up the commits, sorry. When you merge the newsletter one,
you should be able to merge the other 2 on top of it. I will keep an eye on
it and fix everything asap but I tested everything manually and it worked
fine.

Thanks, it was fun working with Python again! I don't think I will have the
time to work on this in the next months (it also does everything I want
now), but I was thinking that it might be relatively easy to turn it into
an modular scraper for free ebooks (think HumbleBundle etc).

niqdev <notifications@github.com> schrieb am Fr., 16. Dez. 2016, 21:28:

**Comment by niqdev** [comment]
Yep, can you just give me a hand to solve the conflicts, I will ask you only to keep an eye on the project when the book will become available again if there is an issue and I might need your help.

I'll be busy as well in the next months, but can you explain me better your idea? Do you know any other "daily free ebook" offers?

**Comment by juzim** [comment]
I will try to get everything working over the weekend.

Not daily but regularly. Lots of indie books or the 1$ base tier on
HumbleBundle. Amazon also has free books (but they are horrible).
Storybundle.com often includes books by authors like Neil Gaiman. I think
even O'Reilly has some free offers.

niqdev <notifications@github.com> schrieb am Fr., 16. Dez. 2016, 22:54:

**Pull request merged**